Userland/Libraries/LibGfx/ImageFormats/GIFLoader.cpp

```diff
@@ -490,10 +490,6 @@ ErrorOr<ImageFrameDescriptor> GIFImageDecoderPlugin::frame(size_t index, Optiona
     frame.image = TRY(m_context->frame_buffer->clone());
     frame.duration = m_context->images[index]->duration * 10;
 
-    if (frame.duration <= 10) {
```

Comment:
https://en.wikipedia.org/wiki/GIF

"with most web browsers automatically displaying the frames with a delay time of 0.1 seconds or more"

https://developers.google.com/speed/webp/docs/riff_container#animation

"Note that the interpretation of the Frame Duration of 0 (and often <= 10) is defined by the implementation. Many tools and browsers assign a minimum duration similar to GIF."

https://wiki.mozilla.org/APNG_Specification#.60fcTL.60:_The_Frame_Control_Chunk

"If the the value of the numerator is 0 the decoder should render the next frame as quickly as possible, though viewers may impose a reasonable lower bound."

So we should do this somewhere. The gif decoder probably isn't the best place, since we should share this logic across gif / apng / webp. Instead, this logic probably belongs at display time (Browser / ImageViewer). But just removing it is I think the wrong thing.

Reply:
Maybe one way to think about this is that we want to interpret a duration of 0 not as 0 but "lasts for one frame".

Reply:
Yeah, that should really not be in the decoder. Moving it to a common place is a good idea, but not for that PR, so I dropped the commit. I admit that I should have look online a little bit more before removing it :sweat_smile:.